classes/Navigation.js

```diff
@@ -0,0 +1,66 @@
+const axios = require('axios');
+const validateStatus = require('../utils/axios-utils')
+
+// Import logger
+const logger = require('../logger/logger');
+
+// Import error
+const { NotFoundError } = require('../errors/NotFoundError')
+
+const GITHUB_ORG_NAME = process.env.GITHUB_ORG_NAME
+const BRANCH_REF = process.env.BRANCH_REF
+
+class Navigation {
+  constructor(accessToken, siteName) {
+    this.accessToken = accessToken
+    this.siteName = siteName
+  }
+
+  async read() {
+    try {
+      const endpoint = `https://api.github.com/repos/${GITHUB_ORG_NAME}/${this.siteName}/contents/_data/navigation.yml`
+
+    const params = {
+      "ref": BRANCH_REF,
+    }
+      
+    const resp = await axios.get(endpoint, {
+      validateStatus,
+      params,
+      headers: {
+          Authorization: `token ${this.accessToken}`,
+          "Content-Type": "application/json"
+        }
+      })
+
+      if (resp.status === 404) throw new NotFoundError ('Navigation page does not exist')
+
+      const { content, sha } = resp.data
+
+      return { content, sha }
+
+    } catch (err) {
+      throw err
+    }
+  }
+
+  async update(newContent, sha) {
+    const endpoint = `https://api.github.com/repos/${GITHUB_ORG_NAME}/${this.siteName}/contents/_data/navigation.yml`
+
+    const params = {
+      "message": 'Edit navigation',
+      "content": newContent,
+      "branch": BRANCH_REF,
+      "sha": sha
+    }
+
+    await axios.put(endpoint, params, {
+      headers: {
+        Authorization: `token ${this.accessToken}`,
+        "Content-Type": "application/json"
+      }
+    })
+  }
+}
+
+module.exports = { Navigation }
```

Comment:
Actually there's no need to create a `Navigation` class, to prevent duplication of code, we can create a new `File` and apply the existing `DataType` type to retrieve the `navigation.yml` file

Reply:
Makes sense, resolved in https://github.com/isomerpages/isomercms-backend/pull/80/commits/e9c5760ba9d38faf01154c164d6107a3e6f22ac2, https://github.com/isomerpages/isomercms-backend/pull/80/commits/4a0b24c018f577edf0de877ed446648ac5f056ed, https://github.com/isomerpages/isomercms-backend/pull/80/commits/a1c294d7968429eae3e85cc2a01ea821fe7d426d